tron/bin/check_tron_jobs.py

```diff
@@ -293,41 +302,110 @@ def guess_realert_every(job):
     return realert_every
 
 
-def compute_check_result_for_job(client, job):
-    kwargs = {
-        "name": "check_tron_job.{}".format(job['name']),
-        "source": "tron",
+def filter_monitoring_config(config):
+    """ Filters out non-Sensu arguments from a job's monitoring config """
+    sensu_args = set([
+        'name',
+        'runbook',
+        'status',
+        'output',
+        'team',
+        'page',
+        'tip',
+        'notification_email',
+        'check_every',
+        'realert_every',
+        'alert_after',
+        'dependencies',
+        'irc_channels',
+        'slack_channels',
+        'ticket',
+        'project',
+        'source',
+        'tags',
+        'ttl',
+        'sensu_host',
+        'component',
+        'description',
+    ])
+
+    include, exclude = {}, {}
+    for k, v in config.items():
+        (include if k in sensu_args else exclude)[k] = v
+    return include, exclude
+
+
+def sort_runs_by_interval(job_content, interval='day'):
+    """ Sorts a job's runs by a time interval (year, month, day, hour, minute,
+    or second), according to a job run's run time.
+    """
+    interval_formats = {
+        'year': '%Y',
+        'month': '%Y.%m',
+        'day': '%Y.%m.%d',
+        'hour': '%Y.%m.%d-%H',
+        'minute': '%Y.%m.%d-%H.%M',
+        'second': '%Y.%m.%d-%H.%M.%S',
     }
-    kwargs.update(job['monitoring'])
-    if 'realert_every' not in kwargs:
-        kwargs['realert_every'] = guess_realert_every(job)
-    kwargs['check_every'] = "{}s".format(_run_interval)
 
-    status = job["status"]
-    if status == "disabled":
-        kwargs["output"] = "OK: {} is disabled and won't be checked.".format(
+    run_buckets = defaultdict(list)
+    if job_content is not None:
+        # Sort runs by shortdate
+        for run in job_content['runs']:
+            # If interval is invalid, will raise a KeyError
+            shortdate = time.strftime(
+                interval_formats[interval],
+                _timestamp_to_timeobj(run['run_time']),
+            )
+            run_buckets[shortdate].append(run)
```

Comment:
Here we are missing a requirement that I neglected to voice.
I want logic that makes buckets for dates that don't even have runs yet, say for the past 7 days.

Right? Seems like if a run doesn't exist at all for a date, then we should alert on that too.

Kinda begs the question of "why do we only go backwards in time for as many days as there are jobruns that exist?

Reply:
I think you can use something similar to guess_realert_every() to decide the number of days for going backward.

Reply:
I've decided to have buckets for all dates from the earliest run we have available to the current time. This way, we ensure that all dates have runs. If there are gaps in our run history (not just in the last 7 days), we will see them.